### ISSUE TITLE
NVDA does not read the message list in Outlook 2016 with the latest Next release

### ISSUE THREAD

**Issue by PratikP1** [nvaccess/nvda#8129]
NVDA throws an error when trying to read the message list in Outlook 2016 with the latest Next release. See version info below. I'm also copying the relevant log to give you a better sense of what errors are being generated. Please note that the  results are the same whether or not all add-ons are disabled.

### Steps to reproduce:
1. Open Outlook 2016.
2. Go to a folder with mail.
3. Use arrow keys to read the message list

### Expected behavior:
Message information such as sender, subject/thread, date/time should be read.

### Actual behavior:
NVDA throws an error tone.

### System configuration:
NVDA version: next-14952,dd1cbe50

NVDA Installed or portable:
Installed

Other information:

Windows version:
Windows 10 1809 build 17128.00. The problem has occurred in previous builds as well.

Name and version of other software in use when reproducing the issue:
Outlook 2016 MSO 16.0.9026.2098

### Other questions:

Does the issue still occur after restarting your PC?
Yes.

Have you tried any other versions of NVDA?
NVDA 2018-1. The problem does not occur.

###  log output

IO - inputCore.InputManager.executeGesture (12:45:22.799):
Input: kb(laptop):alt+tab
IO - speech.speak (12:45:23.016):
Speaking [LangChangeCommand (u'en_US'), u'Inbox - ... - Outlook  row 1  column 2']
IO - speech.speak (12:45:23.997):
Speaking [LangChangeCommand (u'en_US'), u'Inbox - ... - Outlook']
IO - speech.speak (12:45:24.127):
Speaking [LangChangeCommand (u'en_US'), u'Table View  table']
ERROR - eventHandler.executeEvent (12:45:24.180):
error executing event: gainFocus on <NVDAObjects.Dynamic_UIAGridRowTreeviewItemUIA object at 0x07CDB7B0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 152, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "C:\Users\Pratik Patel\AppData\Roaming\nvda\addons\wintenApps\globalPlugins\wintenObjs\__init__.py", line 222, in event_gainFocus
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 961, in event_gainFocus
  File "NVDAObjects\behaviors.pyc", line 471, in reportFocus
  File "NVDAObjects\__init__.pyc", line 849, in reportFocus
  File "speech.pyc", line 383, in speakObject
  File "speech.pyc", line 275, in speakObjectProperties
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "appModules\outlook.pyc", line 451, in _get_name
  File "comtypes\__init__.pyc", line 279, in __getattr__
ValueError: NULL COM pointer access
IO - inputCore.InputManager.executeGesture (12:45:26.696):
Input: kb(laptop):downArrow
ERROR - eventHandler.executeEvent (12:45:26.812):
error executing event: gainFocus on <NVDAObjects.Dynamic_UIAGridRowTreeviewItemUIA object at 0x04155FD0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 152, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "C:\Users\Pratik Patel\AppData\Roaming\nvda\addons\wintenApps\globalPlugins\wintenObjs\__init__.py", line 222, in event_gainFocus
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 961, in event_gainFocus
  File "NVDAObjects\behaviors.pyc", line 471, in reportFocus
  File "NVDAObjects\__init__.pyc", line 849, in reportFocus
  File "speech.pyc", line 383, in speakObject
  File "speech.pyc", line 275, in speakObjectProperties
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "appModules\outlook.pyc", line 451, in _get_name
  File "comtypes\__init__.pyc", line 279, in __getattr__
ValueError: NULL COM pointer access
IO - inputCore.InputManager.executeGesture (12:45:27.969):
Input: kb(laptop):downArrow
ERROR - eventHandler.executeEvent (12:45:28.098):

**Comment by josephsl** [comment]
Hi, does restarting NVDA (with and without add-ons) fix the problem in the short term? Thanks.

**Comment by PratikP1** [comment]
@josephsl wrote: 
> does restarting NVDA (with and without add-ons) fix the problem in the short term?

It does not. Please see my full description. I accidentally cut off my original report. I just updated the comment.

**Comment by leonardder** [comment]
I hoped @bramd was the only one with this issue, turns out he's not. It is another incarnation of an issue @josephsl reported around a week ago. It is currently under investigation what's causing this, pobably an uia bug we need to work aroud.

> Op 27 mrt. 2018 om 19:09 heeft PratikP1 <notifications@github.com> het volgende geschreven:
>

**Comment by leonardder** [comment]
I'm trying to tackle this issue with @bramd, but it causes me headaches so far. Using UIAElement.FindAllBuildCache instead of buildUpdatedCache doesn't solve the issue either.

**Comment by PratikP1** [comment]
Thank you, @leonardder. I look forward to testing it when its incubating in next.

**Comment by ehollig** [comment]
Funny. This only occurs in the Inbox. If I go to another folder like Sent mail for example, it works fine.
With addons disabled, here is the Traceback:
ERROR - eventHandler.executeEvent
error executing event: gainFocus on <NVDAObjects.Dynamic_UIAGridRowTreeviewItemUIA object at 0x06EE6490> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 152, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\__init__.pyc", line 961, in event_gainFocus
  File "NVDAObjects\behaviors.pyc", line 471, in reportFocus
  File "NVDAObjects\__init__.pyc", line 849, in reportFocus
  File "speech.pyc", line 383, in speakObject
  File "speech.pyc", line 275, in speakObjectProperties
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 115, in _getPropertyViaCache
  File "appModules\outlook.pyc", line 454, in _get_name
  File "comtypes\__init__.pyc", line 279, in __getattr__
ValueError: NULL COM pointer access

**Comment by PratikP1** [comment]
@ehollig commented: 
> Funny. This only occurs in the Inbox. If I go to another folder like Sent mail for example, it works fine.

I believe @leonardder  has discovered that the issue is related to the conversation view in inbox and other such folders.

**Comment by PratikP1** [comment]
#7949 testing shows that this issue has been fixed. Thank you, leonardder.

**Comment by PratikP1** [comment]
#7949 testing shows that this issue has been fixed. Thank you, @leonardder.

**Comment by leonardder** [comment]
Thanks. I will close this issue.
